build/download.js

```diff
@@ -0,0 +1,493 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+// @ts-check
```

Comment:
Basically moved all of the `download.js` file from `lib/download.js` into this file.
Most of the code was only for publish phase